code/string-calculator/src/main/java/ru/unn/agile/calc/model/StringCalc.java

```diff
@@ -0,0 +1,121 @@
+package ru.unn.agile.calc.model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class StringCalc {
+    public StringCalc() {
+
+    }
+
+    public double result(final String string) {
+        try {
+            char firstSymbol = string.charAt(0);
+            String[] parsedNumbers = string.split("\\+|\\-|\\*|\\/");
+            String[] parsedOperators = string.split("[0-9\\.]*");
+            int index = 0;
+            char subSymbol = Operation.SUBTRACT.value().charAt(0);
+            if (firstSymbol == subSymbol) {
+                index++;
+                parsedNumbers[index] = subSymbol + parsedNumbers[index];
+                parsedOperators[0] = "";
+            }
+
+            List<Operation> operationsList = new ArrayList<Operation>();
+
+            for (String operator : parsedOperators) {
+                if (Operation.getOperationBySymbol(operator) != null) {
+                    operationsList.add(Operation.getOperationBySymbol(operator));
+                }
+            }
+
+            double[] numbersArray = new double[parsedNumbers.length - index];
+            parseNumbersFromStringToDouble(parsedNumbers, index, numbersArray);
+
+            Operation[] operationsArray = new Operation[operationsList.size()];
+            operationsList.toArray(operationsArray);
+
+            return calcKernel(numbersArray, operationsArray);
+        } catch (NumberFormatException e) {
```

Comment:
It's not a good idea to cover the whole code and catch only NumberFormatException.
What will be in case if we get the String = null or string will be empty ""?

Reply:
Also add a corresponded test.

Reply:
Thank you, test cases have been expanded and try/catch logic has been changed.